VisualizationBase/src/Scene.cpp

```diff
@@ -175,24 +177,48 @@ bool Scene::switchToView(const QString viewName)
 	return view != nullptr;
 }
 
-void Scene::addViewItem(ViewItem *view)
+void Scene::addViewItem(ViewItem *view, QPoint position)
 {
 	Q_ASSERT(!inAnUpdate_);
-	viewItems_.append(view);
+	if (position.x() <= 0 || position.y() <= 0)
+		position = nextEmptyPosition();
+	if (viewItems_[position.x()].size() <= position.y())
+		viewItems_[position.x()].resize(position.y() + 1);
+	//If there already is an item -> use insert
+	if (viewItems_[position.x()][position.y()])
+		viewItems_[position.x()].insert(position.y(), view);
+	//Else just overwrite the nullptr
+	else viewItems_[position.x()][position.y()] = view;
 	addTopLevelItem(view, false);
 }
 
-ViewItem* Scene::newViewItem(const QString name)
+ViewItem* Scene::newViewItem(const QString name, QPoint position)
 {
 	auto result = new ViewItem(nullptr, name);
-	addViewItem(result);
+	addViewItem(result, position);
 	return result;
 }
 
+QPoint Scene::nextEmptyPosition()
```

Comment:
I feel like we're putting too many `ViewItem` related methods in `Scene`. Could you please think of a few different ways to move this functionality away from `Scene` and we can discuss it next time. E.g. we could make these static methods in `ViewItem`, or alternatively, make a `ViewItemManager` which the scenes owns.